python/perf-kernels/fused_moe/moe-gemm.py

```diff
@@ -105,14 +148,27 @@ def moe_gemm_kernel(
             a = tl.load(a_ptrs, mask=(token_mask[:, None] & (offs_k[None, :] < K - k * BLOCK_SIZE_K)), other=0.0)
             b = tl.load(b_ptrs, mask=(offs_k[:, None] < K - k * BLOCK_SIZE_K), other=0.0)
 
-        accumulator = tl.dot(a, b, acc=accumulator)
+        if use_int8_w8a16:
+            accumulator = tl.dot(a, b.to(Out.dtype.element_ty), acc=accumulator)
```

Comment:
I think in general `Out` and `a` should have the same type, but it looks better semantically if `b` is casted to `a`'s type.

Reply:
Right, it makes more sense this way. The dot product will be casted to the out type at the end anyway